partcad/src/partcad/part_factory_scad.py

```
@@ -86,3 +89,108 @@ async def instantiate(self, part):
+                if ".partcad/git" in base_dir.replace("\\", "/"):
+                    # Create a stable name for the SCAD file
+                    hashed = hashlib.sha1(base_dir.encode("utf-8")).hexdigest()
+                    original_filename = os.path.basename(self.path)
+                    new_filename = f"partcad_{hashed}_{original_filename}"
+                    new_scad_path = os.path.join(base_dir, new_filename)
+
+                    print(f"Detected '.partcad/git' in path. Copying to {new_scad_path} ...")
+                    shutil.copy(self.path, new_scad_path)
+
+                    # Read the content
+                    with open(new_scad_path, "r") as f:
```

Ruff (0.8.2)

Unnecessary open mode parameters

Remove open mode parameters

(UP015)